src/gui_common/CustomPopupMenu.cs

```diff
@@ -46,9 +45,43 @@ protected virtual void ResolveNodeReferences()
     {
     }
 
+    /// <summary>
+    ///   Calculates the size the pop should be taking into account minimum
+    ///   height needed by content
+    /// </summary>
+    /// <returns>Returns what should be the size of popup considering content height</returns>
+    protected virtual Vector2 CalculateSize()
+    {
+        RectSize = cachedMinSize;
+
+        var clipControl = GetNodeOrNull<Control>("Panel/Control");
+        var clipControlHeightMargin = 0.0f;
+        if (clipControl != null)
+        {
+            clipControlHeightMargin =
+                Mathf.Abs(clipControl.MarginTop) +
```

Comment:
Also could you explain how the math works out correctly if the margins are negative and this takes the absolute values?

Reply:
The basics from what I can remember is that both negative and positive margins factor into the parent nodes size as absolute / positive values if you look in the Godot editor. I can not remember why at this time, but I'll look into it and try to provide an explanation tomorrow when I'm not sleepy.

Reply:
That would be better, yes. If the clip control is entirely meant to clip the content, then maybe that is the root cause why the popup doesn't size itself correctly automatically?

Reply:
So, looking into it here is the logic for why:

Putting a negative margin on a node removes so many units from the size, however to keep the size of the parent the same Godot then adds that many units to the parent node. Given in this situation the size of the child-most node of CustomPopupMenu is the one that is being considered for the base size of the content we have to manually add the positive and negative margins manually to the total size to get the parent nodes approximate size rather than relying on Godot to do this calculation for us. 

Hopefully that makes sense.

Reply:
I think that does kind of make sense. Could you add a code comment with this info so that someone reading the code later can also get this information?

Reply:
Yep, will do.